src/Systems/MultiplayerSystem.cpp

```diff
@@ -515,61 +583,215 @@ void MultiplayerSystem::pollMovement()
+    message.set_variant(comm::MovementVariant::MAP_UPDATE);
+    *message.mutable_map_positions_update() = mapUpdate;
 
-            const float eps = 0.01f;
-            if (!(std::abs(transformComponent.velocity.x - x) > eps ||
-                  std::abs(transformComponent.velocity.y - y) > eps || transformComponent.scale.x != r))
-                return;
+    auto serializedMessage = message.SerializeAsString();
+
+    m_udp_socket.send(boost::asio::buffer(serializedMessage));
+}
+
+// TODO w przyszłości będzie też zapytanie o stworznie spawnerów
+void MultiplayerSystem::sendSpawnerPosition(
+    comm::StateUpdate& stateUpdate,
+    const std::vector<std::pair<Entity, sf::Vector2<float>>>& spawners)
+{
+    std::sort(m_spawners.begin(), m_spawners.end(),
+              [](const std::pair<Entity, sf::Vector2<float>>& a, const std::pair<Entity, sf::Vector2<float>>& b)
+              { return a.second.x > b.second.x; });
+
+    for (const auto& spawner : spawners)
+    {
+        comm::Enemy* spawnerPosition = stateUpdate.add_enemy_spawner_positions();
+        spawnerPosition->set_position_x(spawner.second.x);
+        spawnerPosition->set_position_y(spawner.second.y);
+        spawnerPosition->set_id(spawner.first);
+    }
+    printf("Sending spawn enemy request\n");
+}
 
-            weaponComponent.pivotPoint.x = newPivotX;
-            weaponComponent.pivotPoint.y = newPivotY;
+void MultiplayerSystem::handleMapUpdate(const google::protobuf::RepeatedPtrField<comm::Enemy>& enemyPositions) const
+{
+    for (const auto& enemy : enemyPositions)
+    {
+        auto gameEntityId = gCoordinator.getGameEntity(enemy.id());
 
-            transformComponent.velocity.x = x - transformComponent.position.x;
-            transformComponent.velocity.y = y - transformComponent.position.y;
-            transformComponent.scale.x = r;
-            weaponTransformComponent.scale.x = r;
+        if (gameEntityId == 0)
+        {
+            continue;
+        }
 
-            colliderComponent.body->SetTransform({convertPixelsToMeters(x), convertPixelsToMeters(y)},
-                                                 colliderComponent.body->GetAngle());
+        if (gCoordinator.hasComponent<TransformComponent>(gameEntityId))
+        {
+            auto& transformComponent = gCoordinator.getComponent<TransformComponent>(gameEntityId);
+            // printf("Enemy's vector x: %f, y: %f\n", enemy.position_x(), enemy.position_y());
+            transformComponent.velocity.x = enemy.position_x() * enemy_speed;
+            transformComponent.velocity.y = -enemy.position_y() * enemy_speed;
         }
     }
 }
 
-void MultiplayerSystem::disconnect()
+std::string MultiplayerSystem::addMessageSize(const std::string& serializedMsg)
 {
-    if (!m_connected) return;
-    m_tcp_socket.close();
-    m_udp_socket.close();
+    comm::BytePrefix prefix;
+    prefix.set_bytes(serializedMsg.size() + DIFF);
+
+    const auto serialisedPrefix = prefix.SerializeAsString();
+
```

Comment:
wydaję mi się że klasa MultiplayerSystem jest źle zrobiony, ma 800 linijek, jest za duży powinien być rozbity na mniejsze systemy, co nie @dominiks01 ?

Reply:
Generalnie jest spory, z dzieleniem jest taki problem że system ma sockety jako
memberów, a większość zdarzeń potrzebuje tych socketów aby się synchronizować.
Inaczej by można w kilku systemach tworzyć aktualizacje w proto a potem w głównym
multiplayer systemie tylko to przetwarzać.